src/main/java/io/kestra/plugin/aws/runner/AwsBatchScriptRunner.java

```diff
@@ -0,0 +1,546 @@
+            containerPropsBuilder.executionRoleArn(runContext.render(this.executionRoleArn));
+        }
+
+        if (commands.getEnv() != null) {
+            containerPropsBuilder
+                .environment(
+                    commands.getEnv().entrySet().stream()
+                        .map(e -> KeyValuePair.builder().name(e.getKey()).value(e.getValue()).build())
+                        .toArray(KeyValuePair[]::new)
+                );
+        }
+
+        String renderedBucket = runContext.render(s3Bucket);
+        String workingDirKey = IdUtils.create();
+        String outputDirKey = IdUtils.create();
+        Map<String, Object> additionalVars = Optional.ofNullable(renderedBucket)
+            .map(bucket -> Map.<String, Object>of(
+                "workingDir", "s3://" + bucket + "/" + workingDirKey,
+                "outputDir", "s3://" + bucket + "/" + outputDirKey
+            ))
+            .orElse(Collections.emptyMap());
+        List<String> command = ScriptService.uploadInputFiles(runContext, runContext.render(commands.getCommands(), additionalVars));
+
+        if (filesToUpload != null && !filesToUpload.isEmpty()) {
+            try (S3TransferManager transferManager = transferManager(runContext)) {
+                filesToUpload.stream().parallel().map(relativePath ->
+                        UploadFileRequest.builder()
+                            .putObjectRequest(
+                                PutObjectRequest
+                                    .builder()
+                                    .bucket(renderedBucket)
+                                    .key(workingDirKey + "/" + relativePath)
+                                    .build()
+                            )
+                            .source(runContext.resolve(Path.of(relativePath)))
+                            .build()
+                    ).map(transferManager::uploadFile)
+                    .map(FileUpload::completionFuture)
+                    .forEach(throwConsumer(CompletableFuture::get));
+            }
+        }
+
+        containerPropsBuilder.command(command)
+            .jobRoleArn(runContext.render(this.jobRoleArn));
+
+        ComputeEnvironmentDetail computeEnvironmentDetail = client.describeComputeEnvironments(
+                DescribeComputeEnvironmentsRequest.builder()
+                    .computeEnvironments(computeEnvironmentArn)
+                    .build()
+            ).computeEnvironments().stream()
+            .findFirst()
+            .orElseThrow(() -> new IllegalArgumentException("Compute environment not found: " + computeEnvironmentArn));
+
+        PlatformCapability platformCapability = switch (computeEnvironmentDetail.computeResources().type()) {
+            case FARGATE:
+            case FARGATE_SPOT:
+                yield PlatformCapability.FARGATE;
+            case EC2:
+            case SPOT:
+                yield PlatformCapability.EC2;
+            default:
+                yield null;
```

Comment:
would it crash in this case?

Reply:
Nope it's what the sdk expects for AKS environment, but as I added a guard in my last commit, it won't ever occur